pyphare/pyphare/pharesee/hierarchy.py

```diff
@@ -179,42 +179,108 @@ def level_range(self):
 
 
 
-def finest_data(pdata, ilvl, hierarchy):
+def are_adjacent(lower_x, upper_x, atol=1e-6):
+    return np.abs(upper_x[0]-lower_x[-1]) < atol
```

Comment:
maybe? 

```
def are_adjacent(lower, upper, atol=1e-6):
    return np.abs(upper-lower) < atol
```

and client callers should do index access I think,

Reply:
ok, will change

Reply:
actually thinking about it, no. the point of that funciton in my opinion is actually that we do not have to think about that logic that makes two sets of coordinates adjacents. If I put upper[0] and lower[-1] at call site, then the function is almost worthless

Reply:
"math.isclose" could be replacing it

Reply:
ok I removed the _x as discussed